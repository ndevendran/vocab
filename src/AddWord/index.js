import React, {Component} from 'react';
import {InputGroup, Button, FormControl} from 'react-bootstrap';

export default class AddWord extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      def: ''
    }

    this.nameOnChange = this.nameOnChange.bind(this);
    this.defOnChange = this.defOnChange.bind(this);
    this.addWord = this.addWord.bind(this);
  }

  nameOnChange(event) {
    this.setState({ name: event.target.value });
  }

  defOnChange(event) {
    this.setState({ def: event.target.value });
  }

  addWord() {
    const name = this.state.name;
    const def = this.state.def;
    this.props.onClick(name, def);
    this.setState({
      name: '',
      def: ''
    });
  }

  render () {
    return (
      <div className="addWord">
        {this.props.hideStep
          ? <div></div>
          :
          <div>
          <InputGroup className="mb-3 wordInput">
            <InputGroup.Prepend>
              <InputGroup.Text>Name</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              onChange={this.nameOnChange}
              value={this.state.name}
            />
          </InputGroup>
            <br />
            <InputGroup className="definitionInput">
              <InputGroup.Prepend>
                <InputGroup.Text>Definition</InputGroup.Text>
              </InputGroup.Prepend>
              <FormControl as="textarea"
                onChange={this.defOnChange}
                value={this.state.def}
              />
            </InputGroup>
            <br />
            <Button variant="primary" onClick={this.addWord}>Add Word</Button>
          </div>
        }
      </div>
    );
  }
}
