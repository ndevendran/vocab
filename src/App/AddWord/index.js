import 'React' from 'react';
import {InputGroup, Button} from 'react-bootstrap';

function AddWord({onClick, nameOnChange, defOnChange, name, def, hideStep}) {
  return (
    <div className="addWord">
      {hideStep
        ? <div></div>
        :
        <div>
        <InputGroup className="mb-3 wordInput">
          <InputGroup.Prepend>
            <InputGroup.Text>Name</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            onChange={nameOnChange}
            value={name}
          />
        </InputGroup>
          <br />
          <InputGroup className="definitionInput">
            <InputGroup.Prepend>
              <InputGroup.Text>Definition</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl as="textarea"
              onChange={defOnChange}
              value={def}
            />
          </InputGroup>
          <br />
          <Button variant="primary" onClick={onClick}>Add Word</Button>
        </div>
      }
    </div>
  );
}
