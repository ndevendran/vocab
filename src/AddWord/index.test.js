import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddWord from '../AddWord';


describe('AddWord', () => {
  const props = {
    onClick: () => true,
    hideStep: false
  }

  it('loads without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<AddWord {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('snapshot test', () => {
    const component = renderer.create(
      <AddWord {...props} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
