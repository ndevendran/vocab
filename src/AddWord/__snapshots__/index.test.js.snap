// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWord snapshot test 1`] = `
<div
  className="addWord"
>
  <div>
    <div
      className="mb-3 wordInput input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          Name
        </span>
      </div>
      <input
        className="form-control"
        onChange={[Function]}
        value=""
      />
    </div>
    <br />
    <div
      className="definitionInput input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          Definition
        </span>
      </div>
      <textarea
        className="form-control"
        onChange={[Function]}
        value=""
      />
    </div>
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add Word
    </button>
  </div>
</div>
`;
